src/target/riscv/riscv.c

```diff
@@ -6031,45 +5798,44 @@ int riscv_init_registers(struct target *target)
 					}
 			}
 
-		} else if (number == GDB_REGNO_PRIV) {
-			sprintf(reg_name, "priv");
+		} else if (reg_num == GDB_REGNO_PRIV) {
 			r->group = "general";
 			r->feature = &feature_virtual;
 			r->size = 8;
 
-		} else if (number >= GDB_REGNO_V0 && number <= GDB_REGNO_V31) {
+		} else if (reg_num >= GDB_REGNO_V0 && reg_num <= GDB_REGNO_V31) {
 			r->caller_save = false;
 			r->exist = (info->vlenb > 0);
 			r->size = info->vlenb * 8;
-			sprintf(reg_name, "v%d", number - GDB_REGNO_V0);
 			r->group = "vector";
 			r->feature = &feature_vector;
 			r->reg_data_type = &info->type_vector;
 
-		} else if (number >= GDB_REGNO_COUNT) {
+		} else if (reg_num >= GDB_REGNO_COUNT) {
 			/* Custom registers. */
+			struct reg_name *additional_regs = info->custom_register_names.reg_names;
+			const unsigned int custom_register_index = reg_num - GDB_REGNO_COUNT;
```

Comment:
It's better to use name `custom_reg_index` -- for consistency with `reg_num`.

```suggestion
			const unsigned int custom_reg_index = reg_num - GDB_REGNO_COUNT;
```

Reply:
Addressed.